cabot_alert_hipchat/models.py

```diff
@@ -9,13 +10,63 @@
+hipchat_template = "Service {{ service.name|safe }} {% if service.overall_status == service.PASSING_STATUS %}is back to normal{% else %}reporting {{ service.overall_status }} status{% endif %}: {{ scheme }}://{{ host }}{% url 'service' pk=service.id %}. {% if service.overall_status != service.PASSING_STATUS %}Checks failing: {% for check in service.all_failing_checks %}{% if check.check_category == 'Jenkins check' %}{% if check.last_result.error %} {{ check.name }} ({{ check.last_result.error|safe }}) {{jenkins_api}}job/{{ check.name }}/{{ check.last_result.job_number }}/console{% else %} {{ check.name }} {{jenkins_api}}/job/{{ check.name }}/{{check.last_result.job_number}}/console {% endif %}{% else %} {{ check.name }} {% if check.last_result.error %} ({{ check.last_result.error|safe }}){% endif %}{% endif %}{% endfor %}{% endif %}{% if alert %}{% for alias in users %} @{{ alias }}{% endfor %}{% endif %}"
 
 # This provides the hipchat alias for each user. Each object corresponds to a User
 class HipchatAlert(AlertPlugin):
     name = "Hipchat"
     author = "Jonathan Balls"
 
+    def _send_images_to_hipchat(self, service):
+        """
+        Find the failing checks for a service and post their images to Hipchat
+        :param service: the failing service
+        :return: None
+        """
+        failing_checks = service.status_checks\
+            .exclude(calculated_status=service.PASSING_STATUS)\
+            .filter(active=True)
+
+        # HIPCHAT_URL has /api/v1 built in, but we need to use v2 for images
+        url = env.get('HIPCHAT_URL').split('v1')[0]
+        url = urljoin(url, 'v2/room/{}/share/file'.format(env.get('HIPCHAT_ALERT_ROOM')))
+        headers = {
+            'Content-type': 'multipart/related; boundary=boundary123456',
+            'Authorization': 'Bearer {}'.format(env.get('HIPCHAT_API_V2_KEY'))
+        }
+        message = {"message": "Upload image for failing status check"}
+
+        for check in failing_checks:
+            image = check.get_status_image()
+            if image is not None:
+                # See https://www.hipchat.com/docs/apiv2/method/share_file_with_room
```

Comment:
I was heavily inspired by the first one (it's basically [the same](https://gist.github.com/bdclark/0cbadce5816b6ab10eb2#file-hipchat_file-py-L37-L50) as what I wrote--I should probably link to it actually.

Looks like the libraries people posted in the stackoverflow link are for api v1, which as far as I can tell doesn't support posting images (except in cards, which seemed unnecessary)